### ISSUE TITLE
implementation of local fully connected layer 

### ISSUE THREAD

**Issue by metalbubble** [BVLC/caffe#1875]
Here I want to construct a local fully connected layer functioning as follows, I am not sure if I could directly use current existing layers to construct this layer, or I have to code my own layer cpp.

Let's say the bottom of this local fully connected layer is the pool5 layer ouput in alexNet as 6x6x256 (256 is the number of filters), the inner product of this local fully connected layer happens to each feature map 6x6 individually, rather than to the vectorized 6x6x256 in a whole, so the output of this local fully connected layer is 1x1x256 (it is possible we have more than 256 output number for sure, it is dependent on the number of nodes at the local FC layer). On the other hand, this local fully connected layer also could be considered as a weighted pooling. But does the weights of this weighted pooling could be learned as the inner product weights in fc layer?

Thanks a lot for the help!

**Comment by ducha-aiki** [comment]
#1271

**Comment by metalbubble** [comment]
@ducha-aiki Thanks for the hint. I checked #1271, but it looks like not what I want. For example, the input is 6x6x256, the local connected layer in #1271 has a different weighted sum on the 256 channels for each one of 6x6 locations. For my case, it should have a different weighted sum on the 6x6 feature map, for each one of the 256 channels. It is other way around to #1271. I figure out a hack way to do it in the network prototxt: I found that there is a parameter in CONVOLUTION as group. Then let's see the bottom layer is 6x6x256, I can add a CONVOLUTION layer on top of it, then set its parameter group:256 and kernel_size:6. So there would be one convolution on one feature map respectively, acting as weighted sum of the feature map value. I didn't test it yet, not sure if it works...
